src/config/contracts.ts

```diff
@@ -23,7 +33,12 @@ const MAINNET_ADDRESSES = {
   Validators: '0xaEb865bCa93DdC8F47b8e29F40C5399cE34d0C58',
 } as const;
 
-export const Addresses = config.isAlfajores ? ALFAJORES_ADDRESSES : MAINNET_ADDRESSES;
+export const Addresses =
+  config.chain.rpcUrls.default.http[0] === celo.rpcUrls.default.http[0]
+    ? MAINNET_ADDRESSES
+    : config.chain.rpcUrls.default.http[0] === celoAlfajores.rpcUrls.default.http[0]
+      ? ALFAJORES_ADDRESSES
+      : UNKNOWN_TESTNET_ADDRESSES;
```

Comment:
what's the idea behind having `UNKNOWN_TESTNET_ADDRESSES` at all?

Reply:
To make sure we can use any network (not alfajores/mainnet) which could have different addresses for these contracts. Since a couple lines below we use the following logic to resolve:
```ts
  if (Addresses[name] === ZERO_ADDRESS) {
    console.log(`Resolving ${name} address from registry`);
```

means i could just have an empty object w/ undefined values